### PR TITLE
Support fast local live migration

### PR DESCRIPTION
**Pull request by rbradford**
Pass the FDs for the guest memory across the socket allowing very fast live migration (and for live upgrade.)

Fixes: #3566

### CODE REVIEW

File: vmm/src/api/mod.rs [hunk 1]
```diff
@@ -195,6 +195,7 @@ pub struct VmReceiveMigrationData {
 pub struct VmSendMigrationData {
     /// URL to migrate the VM to
     pub destination_url: String,
+    pub local: bool,
```
Comment (likebreath) [inline vmm/src/api/mod.rs:198]: Please don't forget to add it to our OpenAPI spec file.

File: vmm/src/lib.rs [hunk 2]
```diff
@@ -1049,6 +1075,10 @@ impl Vmm {
             })?
         };
 
+        if send_data_migration.local {
```
Comment (likebreath) [inline vmm/src/lib.rs:1078]: Looks like we will need to update our vm-migration protocol for this change (e.g. the comments on the top of `vm-migration/src/protocol.rs`).

**Review by likebreath — COMMENTED** [review]
LGTM. Given, all our live-migration tests are actually in "local mode", we should be able to easily expand them to test this patch.

File: vmm/src/api/mod.rs (hunk 1 above)
Comment (sboeuf) [reply vmm/src/api/mod.rs:198]: Add a comment to explain this attribute's purpose.

**Review by sboeuf — APPROVED** [review]
Looking great :+1:

**Pull request merged**